generator/resources/ec2_linux_test_matrix.json

```diff
@@ -54,6 +54,39 @@
     "binaryName": "amazon-cloudwatch-agent.rpm",
     "family": "linux"
   },
+  {
+    "os": "al2",
+    "username": "ec2-user",
+    "instanceType":"m7g.medium",
+    "installAgentCommand": "go run ./install/install_agent.go rpm",
+    "ami": "cloudwatch-agent-integration-test-arm64-al2*",
+    "caCertPath": "/etc/ssl/certs/ca-bundle.crt",
+    "arc": "arm64",
+    "binaryName": "amazon-cloudwatch-agent.rpm",
+    "family": "linux"
+  },
+  {
+    "os": "al2023",
+    "username": "ec2-user",
+    "instanceType":"t3a.medium",
+    "installAgentCommand": "go run ./install/install_agent.go rpm",
+    "ami": "cloudwatch-agent-integration-test-x86-al2023*",
+    "caCertPath": "/etc/ssl/certs/ca-bundle.crt",
+    "arc": "amd64",
+    "binaryName": "amazon-cloudwatch-agent.rpm",
+    "family": "linux"
+  },
+  {
+    "os": "al2023",
+    "username": "ec2-user",
+    "instanceType":"m7g.medium",
+    "installAgentCommand": "go run ./install/install_agent.go rpm",
+    "ami": "cloudwatch-agent-integration-test-aarch64-al2023*",
```

Comment:
Closing the comment. Talked with @ymtaye  and this is intended to avoid mismatching with the pattern of `cloudwatch-agent-integration-test-arm64-al2` which collides with AL2 AMI.